backends/aoti/runtime/AotiBackend.cpp

```diff
@@ -0,0 +1,532 @@
+
+    // Load the ELF using dlopen
+    void* so_handle = dlopen("/tmp/test.so", RTLD_LAZY | RTLD_LOCAL);
+    if (so_handle == nullptr) {
+        std::cout<<dlerror()<<std::endl;
+        return Error::AccessFailed;
+    }
+
+    AOTInductorModelContainerCreateWithDevice = 
+      reinterpret_cast<AOTInductorModelContainerCreateWithDeviceFunc>(
+        dlsym(so_handle, "AOTInductorModelContainerCreateWithDevice")
+      );
+    if (AOTInductorModelContainerCreateWithDevice == nullptr) {
+      perror("dlsym1");
+      return Error::AccessFailed;
+    }
+    AOTInductorModelContainerDelete = 
+      reinterpret_cast<AOTInductorModelContainerDeleteFunc>(
+        dlsym(so_handle, "AOTInductorModelContainerDelete")
+      );
+    if (AOTInductorModelContainerDelete == nullptr) {
+      perror("dlsym2");
+      return Error::AccessFailed;
+    }
+    AOTInductorModelContainerGetNumInputs = 
+      reinterpret_cast<AOTInductorModelContainerGetNumInputsFunc>(
+        dlsym(so_handle, "AOTInductorModelContainerGetNumInputs")
+      );
+    if (AOTInductorModelContainerGetNumInputs == nullptr) {
+      perror("dlsym3");
+      return Error::AccessFailed;
+    }
+    AOTInductorModelContainerGetNumOutputs = 
+      reinterpret_cast<AOTInductorModelContainerGetNumOutputsFunc>(
+        dlsym(so_handle, "AOTInductorModelContainerGetNumOutputs")
+      );
+    if (AOTInductorModelContainerGetNumOutputs == nullptr) {
+      perror("dlsym4");
+      return Error::AccessFailed;
+    }
+    AOTInductorModelContainerRun = 
+      reinterpret_cast<AOTInductorModelContainerRunFunc>(
+        dlsym(so_handle, "AOTInductorModelContainerRun")
+      );
+    if (AOTInductorModelContainerRun == nullptr) {
+      perror("dlsym5");
+      return Error::AccessFailed;
+    }
+
+    AOTInductorModelContainerHandle container_handle = nullptr;
+
+    AOTIRuntimeError err;
+
+    err = AOTInductorModelContainerCreateWithDevice(
+      &container_handle,
+      1,
+      "cuda",
+      nullptr);
+    printf("container_handle=%p\n", container_handle);
+
+    AOTIDelegateHandle* handle = new AOTIDelegateHandle();
+    handle->so_handle = so_handle;
+    handle->container_handle = container_handle;
+    return (DelegateHandle*)handle; // Return the handle post-processing
+  }
+
+  // Once per execution
+  Error execute(
+    BackendExecutionContext& context,
+    DelegateHandle* handle_,
+    EValue** args) const override {
+    AOTIDelegateHandle* handle = (AOTIDelegateHandle*)handle_;
+
+    size_t num_inputs;
+    AOTInductorModelContainerGetNumInputs(
+      handle->container_handle,
+      &num_inputs);
+
+    size_t num_outputs;
+    AOTInductorModelContainerGetNumOutputs(
+      handle->container_handle,
+      &num_outputs);
+
+    std::vector<AOTITensorHandle> inputs(num_inputs);
+    std::vector<AOTITensorHandle> outputs(num_outputs);
+
+    for (int i = 0; i < num_inputs; i++) {
+      auto tensor_in = args[i]->toTensor();
+      inputs[i] = &tensor_in;
```

Comment:
this has UB as written. Needs to be

```
const auto& tensor_in = args[i]->toTensor();
inputs[i] = &tensor_in;
```

or better yet

```
inputs[i] = &args[i]->toTensor();
```

as written, it copies the Tensor out of the EValue and creates a dangling pointer.